### PR TITLE
New AsyncAPI naming

### LINKED ISSUE #638: Fix faststream docs gen producing wrong asyncapi
Publishers and subscribers in the doc should be named after the topics they are connected to, and that is not the case in the generated docs
The topic name is described in the "channel specific info"

Schema is generated in https://github.com/airtai/faststream/blob/main/faststream/kafka/asyncapi.py#L31

### PR DESCRIPTION
**Pull request by Sternakt**
# Description

Rework the asyncapi.py for Kafka and Rabbit brokers to now name the channels by their respective topics.

Fixes #638 

## Type of change

- [x] Bug fix (a non-breaking change that resolves an issue)
- [ ] This change requires a documentation update

## Checklist

- [x] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [x] I have conducted a self-review of my own code
- [x] I have made the necessary changes to the documentation
- [x] My changes do not generate any new warnings
- [x] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [x] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [x] I have ensured that static analysis tests are passing by running `scripts/static-anaylysis.sh`
- [ ] I have included code examples to illustrate the modifications

### CODE REVIEW

**Comment by Lancetnik** [comment]
I can't approve these changes and I am totally disagree with this naming policy.

**Comment by Lancetnik** [comment]
According AsyncAPI channel name is an operation name, not the subject one. All subject information should contains in bindings. This reason I suggest to name our channels based on the handler functions name.
Adding topic name to it - just a my initiatives, but it conflicts with the original AsyncAPI spec. For now you are totally breaks this logic.

**Comment by Lancetnik** [comment]
Also, before so breaking changes, we need to fix them in the our specification, write some function -> names examples, conform our view, and then write a code

**Comment by Lancetnik** [comment]
I think, we need to ask some help from AsyncAPI group to resolve our view

**Review by Lancetnik — CHANGES_REQUESTED** [review]
We need to represent and approve our view before code writing

**Comment by davorrunje** [comment]
Let's discuss it all together on Discord when you have time

File: faststream/kafka/asyncapi.py [hunk 1]
```diff
@@ -86,27 +72,24 @@ class Publisher(LogicPublisher, AsyncAPIOperation):
...
             call_model = build_call_model(call)
             body = get_response_schema(
                 call_model,
-                prefix=to_camelcase(call_model.call_name),
+                prefix=self.topic + ":Message:",
             )
             if body:
                 payloads.append(body)
-
+        print(payloads)
```
Comment (Lancetnik) [inline faststream/kafka/asyncapi.py:85]: Remove all debug prints. please

Comment (Sternakt) [reply faststream/kafka/asyncapi.py]: Removed

**Comment by Lancetnik** [comment]
Btw, can you change Nats naming behavior too and write AsyncAPI tests for it?

**Comment by Sternakt** [comment]
> Btw, can you change Nats naming behavior too and write AsyncAPI tests for it?

Done :)

**Comment by Lancetnik** [comment]
@Sternakt looks like `Msg` in payload naming is needless `test_test:Handle:Message:Msg:Payload`

**Comment by Lancetnik** [comment]
Also, we have no tests on Publisher naming, I'll add it

**Review by Lancetnik — APPROVED** [review]

**Comment by Lancetnik** [comment]
@Sternakt, @davorrunje, 0.2.3 is ready to merge

**Review by davorrunje — APPROVED** [review]

**Pull request merged**